### ISSUE TITLE
As an FS Administrator I need to be able to edit the dates and times for Christmas tree sales and season for my forest.

### ISSUE THREAD

**Issue by ASprinkle** [nciinc/fs-permit-platform#247]
## Acceptance Criteria
- [ ]

## Definition of Done
- [ ] Code unit / e2e tested and comply with QWASP
- [ ] No new security vulnerabilities
- [ ] Internal team code review
- [ ] Documentation / readme.md / CircleCI updated
- [ ] Docker updated
- [ ] Compare finished design with mockup
- [ ] Usability tested

**Comment by ASprinkle** [comment]
I do not think we need this story any more if we intend to create changes via a request form story #316

**Comment by lauraGgit** [comment]
for the user auth - one thing we might try is to update the whitelist from an array of email strings to an array of objects - with `user_email` and `forest` rolls. So that it would look something like this:
[{'user_email': 'eauth_authed_email', 'forest': 'forest_shortname'}, {'user_email': 'eauth_authed_email', 'forest': 'all'}] as a start.

**Comment by sadlerw** [comment]
@bryanmcfadden here is the design of the page.  I just copied the existing design.
![Screen Shot 2018-02-08 at 8.26.50 AM.png](https://images.zenhubusercontent.com/595e640361b4ca3b0a7370e7/f45af1f6-ba5e-49ee-b7d7-0d3884b3300d)

**Comment by sadlerw** [comment]
here is the alert after success

![Screen Shot 2018-02-08 at 8.28.30 AM.png](https://images.zenhubusercontent.com/595e640361b4ca3b0a7370e7/4700d6af-2215-4ca2-b70c-5074c23db6ea)

**Comment by ASprinkle** [comment]
The update bar showing the new season dates is in YYYY-MM-DD format. Is that what the FS administrator sees? Or is that format displayed for the public?  While I understand it, I wonder if people will get confused by the presentation of YYYY-MM-DD, instead of writing it out (November 01, 2018) or some other format

**Comment by sadlerw** [comment]
We could reformat that to look like what is on the guidelines - right now it just matches what is in the inputs below.  I'll add something.

**Comment by JaneZC** [comment]
I just had a content-updating type of question as it relates to Story 316.                   

If the website season dates are being updated using the format/method in this story (#247), do we need season dates to be included as content that can/should be updated via the Google form that's to be developed as part of #316?

Also ... just a general concern that might not apply here but I wanted to at least mention ... I would think it best to have just one method for content administrators to have to learn to update content on a site (if possible). The thinking would be that it easier to only have to learn one content-updating method ... instead of having to learn two ... and, after learning two methods, then folks will also have to remember which method updates which particular pieces of content on a site. (I should say, though, the date form looks super easy!!)

Its very possible I am not following the use cases here and am not privvy to all the needed information. (For example, perhaps different people will be updating different pieces of content?) But, I just wanted to mention.

**Comment by ASprinkle** [comment]
The google form story is so that FS admins can make edit requests for general information on the pages. 
This story is related to season sales dates and times which we know will have to be updated annually.  This story is supposed to allow the FS admin to make the change directly w/o having to request and wait.  I am less concerned about FS admins having to be familiar with two different methods for making updates.

**Comment by hannahkane** [comment]
Given @ASprinkle's comment, I'm wondering if we need a design that shows what the form looks like if the user only has permissions to edit a single forest. Do they still see the pulldown of forests?

**Comment by sadlerw** [comment]
good question @hannahkane   In the past I've seen the dropdown modified or removed when only one selection is present.  However, in general it creates both simpler and easier to maintain code and provides a more consistent user experience if you keep the dropdown even with only one entry.  The idea is that at some point if the user gets another entry the control doesn't suddenly switch from a label to a dropdown.   

One solution I've also seen is to pre-select the first option if there is only one.  Then the user can skip changing it.  But that means that every user would have the first option pre-selected even if they have more than one.  The big drawback is that if the user does have more than one and the wrong one is selected they never know!  If they have to select they realize that there is an issue when they don't select or at least by selecting they know they have picked the right thing.

All that said, for 4 forests we should probably be using radio buttons with the first (or only) forest selected.  Ultimately there will be more than 4 forests and this will have to be a typeahead or search which looks more like a dropdown, plus it uses less screen real estate, so I'm torn.

**Comment by npage123** [comment]
I like what @sadlerw is proposing. But I think we'll need to change the "To update season dates, please select a forest." to possibly "To update season dates, please select a forest you have access to." or something along those lines. Any suggestions @JaneZC ?

**Comment by sadlerw** [comment]
It could stay the same but we add "If the forest you wish to change is not present, please request access" where request access is a link to the google doc for requesting changes?

**Comment by sadlerw** [comment]
so something like:

<img width="823" alt="screen shot 2018-02-12 at 11 51 54 am" src="https://user-images.githubusercontent.com/2058839/36111276-35afd542-0feb-11e8-8eae-862a7173d351.png">

**Comment by JaneZC** [comment]
@sadlerw @npage123 We could make a small change ... "To update season dates, please select your forest." 

Presumably, administrators will be coming to this page with a specific task in mind and only for their forest.

Also, if the forest is unlisted (so not one of the four Christmas tree pilot forests), then there shouldn't be any content to change because it doesn't exist? And/or, would administrators for those unlisted forests have access to this part of the module? (I am not sure how this section fits into the IA as it relates to the parent module?)

**Comment by JaneZC** [comment]
@sadlerw Thank you for explaining! To recap, the person who visits this page will only see forests listed that they currently have permissions to see and/or request changes too. (I had mistakenly assumed that all four pilot forests would be visible and featured in the radio buttons.)

I would change the line to:

If your forest is not listed, please submit this form to request access.

Also, it may be helpful to add a divider between the hyperlink to the form and the line that starts "The time on the season .... ".

**Comment by sadlerw** [comment]
reports page will change to:
<img width="937" alt="screen shot 2018-02-12 at 12 53 13 pm" src="https://user-images.githubusercontent.com/2058839/36113827-bd155c8e-0ff3-11e8-8bac-080ca0b4ec60.png">

**Comment by tram** [comment]
Looks good! The forest selection should likely be changed from radio buttons to a dropdown when the project moves beyond a pilot.

**Comment by sadlerw** [comment]
yea it will most likely have to be some sort of search control like a typeahead/autocomplete eventually

**Comment by ASprinkle** [comment]
@tram @sadlerw  Are radio buttons sufficient for what we're trying to accomplish with the pilot? It all looks really good to me.  Is there any need to change it from radio buttons to a drop down now, that would help the pilot be successful?

**Comment by sadlerw** [comment]
We originally had it as a typeahead but couldn't make that pass accessibility.  I think it's fine to kick it down the road to a later project when perhaps an accessible version of the typeahead is available.  Or maybe even later in this project.
